pyquil_for_azure_quantum/__init__.py

```diff
@@ -196,30 +276,35 @@ def __init__(self, *, target: str) -> None:
             name=target,
         )
 
-    def execute(  # type: ignore[override]
+    def execute(
         self,
-        executable: AzureProgram,
+        # See the comment on the `AzureQuantumMachine` class for why the type error is ignored.
+        executable: AzureProgram,  # type: ignore[override]
         memory_map: Optional[MemoryMap] = None,
-        name: str = "pyquil-azure-job",
-        **_kwargs: Any,  # unused, but defined here to match QAM superclass.
+        **kwargs: Any,  # used for `name`, to ensure signature compatibility with QAM
```

Comment:
I expected `# type: ignore[override]` to let me ignore new arguments, especially since the parent already has `kwargs` and `name` was optional (so this isn't actually an override incompatibility at all), but that didn't work. Arguably this is a bug in `mypy`.